src/util/ringdelaybuffer.cpp

```diff
@@ -0,0 +1,160 @@
+#include "ringdelaybuffer.h"
+
+RingDelayBuffer::RingDelayBuffer(SINT bufferSize)
+        : m_bufferSize(bufferSize),
+          m_readPos(0),
+          m_writePos(0),
+          m_ringFullMask((bufferSize * 2) - 1),
+          m_ringMask(bufferSize - 1),
+          m_jumpLeftAroundMask(0),
+          m_buffer(bufferSize) {
+    // TODO(davidchocholaty) Handle to allow only power of two for the size of the ring delay buffer.
+
+    // Set the ring delay buffer items to 0.
+    CSAMPLE* bufferData = m_buffer.data();
+    memset(bufferData, 0, sizeof(*bufferData) * bufferSize);
```

Comment:
Please avoid invasive manipulation like this. Manual handling of sizes and memsetting is prone to cause buffer overflow problems. 
```suggestion
    m_buffer.fill(0);
```